agents/chopper_mc2000b/chopper_mc2000b_agent.py

```diff
@@ -0,0 +1,339 @@
+                            'externalouter': 2,
+                            'externalinner': 3}
+
+
+class ControllerAgent:
+    """Agent to connect to the MC2000B Thorlabs chopper controller
+    device.
+
+    Parameters
+    __________
+
+    comport : str
+        COM port to connect to device. Ex: "COM3"
+    nbaud : int
+        baud rate of the device (115200)
+    timeout : int
+        the timeout time for the device; default is set to 3s
+    """
+
+    def __init__(self, agent, comport, nbaud=115200, timeout=3):
+        self.agent = agent
+        self.log = agent.log
+        self.lock = TimeoutLock()
+        self.comport = comport
+        self.nbaud = nbaud
+        self.timeout = timeout
+
+        self.hdl = MC2000BOpen(self.comport, self.nbaud, self.timeout)
+
+        self.initialized = False
+        self.take_data = False
+
+        agg_params = {'frame_length': 60,
+                      'exclude_influx': False}
+
+        # register the feeds
+        self.agent.register_feed('input_freqs',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=1
+                                 )
+
+        self.agent.register_feed('output_freqs',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=1
+                                 )
+
+    @ocs_agent.param('auto_acquire', default=False, type=bool)
+    def init_chopper(self, session, params):
+        """init_chopper(auto_acquire=False)
+
+        **Task** - Perform first time setup of MC2000B chopper controller
+        communication.
+
+        Parameters
+        __________
+
+        auto_acquire : bool, optional
+            Default is false. Starts data acquisition after
+            initilialization if True.
+        """
+        if self.initialized:
+            return True, "Already initialized"
+
+        with self.lock.acquire_timeout(job='init_chopper') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start Task because "
+                              f"{self.lock.job} is already running")
+                return False, "Could not acquire lock"
+
+            session.set_status('running')
+
+            # Establish connection to the chopper controller
+            self.hdl = MC2000BOpen(self.comport, self.nbaud, self.timeout)
+
+        if (self.hdl == 0):
+            self.initialized = True
+            self.log.info("Chopper connected")
+        else:
+            self.log.warn("Chopper not connected")
+
+        # Start data acquisition if requested
+        if params['auto_acquire']:
+            resp = self.agent.start('acq', params={})
+            self.log.info(f'Response from acq.start(): {resp[1]}')
+
+        return True, "Chopper controller agent initialized"
+
+    @ocs_agent.param('freq', type=int)
```

Comment:
Does the set frequency of the chopper have to be an `int`? Or can it be a `float`?

Reply:
I just double-checked with Joe. It depends on the reference point you set for the chopper blade (and thus, the bladetype). If you've got a blade where you have small and large slits, you can pass floats for the small slits but not for the large ones.